aspnetcore/tutorials/web-api-vsc.md

```diff
@@ -0,0 +1,272 @@
+
+Add a folder named *Models*. You can put model classes anywhere in your project, but the *Models* folder is used by convention.
+
+Add a `TodoItem` class with the following code:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoItem.cs)]
+
+* The `[Key]` data annotation denotes the property, `Key`, is a unique identifier.
+* `[DatabaseGenerated` specifies the database will generate the key (rather than the application).
+* `DatabaseGeneratedOption.Identity` specifies the database should generate integer keys when a row is inserted. 
+
+### Create the database context
+
+The *database context* is the main class that coordinates Entity Framework functionality for a given data model. You create this class by deriving from the `Microsoft.EntityFrameworkCore.DbContext` class.
+
+Add a `TodoContext` class in the *Models* folder:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoContext.cs)]
+
+## Add a repository class
+
+A *repository* is an object that encapsulates the data layer. The *repository* contains logic for retrieving and mapping data to an entity model. Create the repository code in the *Models* folder.
+
+Defining a repository interface named `ITodoRepository`:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/ITodoRepository.cs)]
+
+This interface defines basic CRUD operations.
+
+Add a `TodoRepository` class that implements `ITodoRepository`:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Models/TodoRepository.cs)]
+
+Build the app to verify you don't have any compiler errors.
+
+## Register the repository
+
+By defining a repository interface, we can decouple the repository class from the MVC controller that uses it. Instead of instantiating a `TodoRepository` inside the controller we will inject an `ITodoRepository` using the built-in support in ASP.NET Core for [dependency injection](xref:fundamentals/dependency-injection).
+
+This approach makes it easier to unit test your controllers. Unit tests should inject a mock or stub version of `ITodoRepository`. That way, the test narrowly targets the controller logic and not the data access layer.
+
+In order to inject the repository into the controller, we need to register it with the DI container. Open the *Startup.cs* file. 
+
+In the `ConfigureServices` method, add the highlighted code:
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Startup.cs?name=snippet_AddSingleton&highlight=11)]
+
+## Add a controller
+
+In the *Controllers* folder, create a class named `TodoController`. Add the following (and add closing braces):
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Controllers/TodoController.cs?name=snippet_todo1)]
+
+This defines an empty controller class. In the next sections, we'll add methods to implement the API.
+
+## Getting to-do items
+
+To get to-do items, add the following methods to the `TodoController` class.
+
+[!code-csharp[Main](first-web-api/sample/TodoApi/Controllers/TodoController.cs?name=snippet_GetAll)]
```

Comment:
I had to download a formatting tool extension. As mentioned, I'm not documenting VS Code.